docs/tutorial/t-spark-monitoring.md

```diff
@@ -0,0 +1,350 @@
+## Monitoring the Spark Cluster
+
+By default, Spark stores the logs of drivers and executors as pod logs in the local file system, which gets lost if the pods are deleted. Spark provides us with the feature to store these logs in S3 so that they can later be retrieved and visualized.
+
+Monitoring of Spark cluster can be done in two ways,
+1. Using Spark History Server 
+2. Using Canonical Observability Stack (COS)
+
+Let's explore each one of these.
+
+### Monitoring with Spark History Server
+
+Spark History Server is a user interface to monitor the metrics and performance of completed and running Spark applications. The Spark History Server is offered as a charm in the Charmed Spark solution, which can be deployed via Juju.
+
+Let's create a fresh Juju model for the experiments with Spark History server.
+
+```bash
+juju add-model history-server
+```
+
+To enable monitoring via Spark History server, we must first create a service account with necessary configurations for Spark jobs to store logs in a S3 bucket. Then we need to deploy Spark History server with Juju and configure it to read from the same S3 bucket that the Spark writes logs to.
+
+Since Juju has already created the namespace `history-server`, let's create a new service account in this namespace. We'll reuse configuration options from the existing `spark` service account and a few more configuration options to instruct Spark pn where to store the event logs.
+
+```bash
+# Get config from old service account and store in a file
+spark-client.service-account-registry get-config \
+  --username spark --namespace spark > properties.conf
+
+# Add a few more config options for the storage of event logs
+echo "spark.eventLog.enabled=true" >> properties.conf
+echo "spark.eventLog.dir=s3a://spark-tutorial/spark-events/" >> properties.conf
+echo "spark.history.fs.logDirectory=s3a://spark-tutorial/spark-events/" >> properties.conf
+
+# Create a new service account and load configurations from the file
+spark-client.service-account-registry create \
+  --username spark --namespace history-server \
+  --properties-file properties.conf
+```
+
+We've configured Spark to write event logs to `spark-events` directory in the `spark-tutorial` bucket. However, that directory doesn't exist yet. Let's create that directory in S3.
+
+```bash
+aws s3api put-object --bucket spark-tutorial --key spark-events/ # / at the end is requied
+```
+
+Now, let's deploy the [`spark-history-server-k8s`](https://github.com/canonical/spark-history-server-k8s-operator) charm in our Juju model. 
+
+```bash
+juju deploy spark-history-server-k8s -n1 --channel 3.4/stable
```

Comment:
Probably I will deploy from `edge
`

Reply:
Actually, ideally, we should produce documentation for stable branches if possible, as edge channels may be unstable and I don't want people testing out things to hit road blocks

Reply:
Keeping up with `3.4/stable` for now.